### DISCUSSION TITLE
Support Link HTTP headers with `rel=preload`

### DISCUSSION THREAD

**Discussion by AKockWorth** [vercel/next.js#26605] [Ideas]
### Describe the feature you'd like to request

If the NextJS web server would, in **addition** to the preload links in the HTML source, also respond with Link HTTP headers, we could use [this Nginx](http://nginx.org/en/docs/http/ngx_http_v2_module.html#http2_push_preload) feature to automatically HTTP/2 Push the resources to the client.

### Describe the solution you'd like

The HTTP handler that renders the HTML should be able to access the information about which scripts to include and set the desired headers.

### Describe alternatives you've considered

A separate proxy web server that parses the HTML, injects the headers, and passes on the response.

**Comment by peteychuk** [comment]
That would be great to have a Link HTTP header together with an HTML link tag in order to optimize performance. It's a very popular approach and for example, [Cloudflare](https://www.cloudflare.com/website-optimization/http2/serverpush/) supports it and enables server push by default, Nginx and [Ingress-nginx ](https://kubernetes.github.io/ingress-nginx/user-guide/nginx-configuration/annotations/#http2-push-preload) allows to enable server push.
Nginx wrote a good blog post "[Introduce HTTP/2 Server Push](https://www.nginx.com/blog/nginx-1-13-9-http2-server-push/)" that explains how to work with Server Push.

**Comment by michalszorad** [comment]
Is there any update on this?
